### DISCUSSION TITLE
Call back URLs and redirect for sign in and new users

### DISCUSSION THREAD

**Discussion by j-lee8** [nextauthjs/next-auth#1809] [Help] [answered]
I feel like there's a lot of black magic happening with this library and I don't understand why it isn't redirecting. I have a few questions;

1. In Google developer console, what should the authorised redirect URLs be?? I have tried the following but no luck:
- `http://localhost:3000/api/auth/callback/google`
- `http://localhost:3000/app/main`

And these are the URIs - what should they be? I am running at localhost:3000
- `http://localhost`
- `http://127.0.0.1`

2. Should I be doing a manual router.push on sign in, or does the callback URL do that for me? Even if I router.push, it just goes from `/login to /login#`, or appends the Google auth redirect to the end of it. I've also tried the absolute URL but still the same. I don't feel like any of this is documented properly. The /login page is located in `pages/login ` and I have configured `[...nextauth].ts` to have:
```
pages: {
        signIn: '/login'
}
```

In my _app.tsx I have:

```
export default function App({ Component, pageProps }) {
	const apolloClient = useApollo(pageProps.initialApolloState);

	return (
		<ApolloProvider client={apolloClient}>
			<Provider session={pageProps.session}>
				{Component.auth ?
					<Auth>
						<MainHeader>
							<Component {...pageProps} />
						</MainHeader>
					</Auth>
					:
					<PageLayout>
						<Component {...pageProps} />
					</PageLayout>
				}
			</Provider>
		</ApolloProvider>
	);
}

function Auth({ children }) {
	const router = useRouter();
	const [session, loading] = useSession();
	const isUser = !!session?.user;

	useEffect(() => {
		if (loading) return; // Do nothing while loading
	  	if (!isUser) router.push('/login'); // If not authenticated, force log in
		if (isUser) router.push('/app/main'); // pages/app/main
	}, [isUser, loading]);

	if (isUser) return children;

	// Session is being fetched, or no user.
	// If no user, useEffect() will redirect.
	return <div>Loading...</div>;
}
```

Something isn't linking up correctly and I really have no idea why. The URLs are correct.

I really hope somebody can help me because this has been bugging me for days now and I want to progress.

Thanks all.

**Comment by j-lee8** [comment, marked as answer]
Sorted it.

**Reply by muhaimincs** [reply]
How did you sorted it
